lib/sync.js

```diff
@@ -613,6 +613,9 @@ SyncApi.prototype._processSyncResponse = function(syncToken, data) {
         client.store.storeAccountDataEvents(events);
         events.forEach(
             function(accountDataEvent) {
+                if (accountDataEvent.getType() == 'm.push_rules') {
+                    client.pushRules = accountDataEvent.getContent();
```

Comment:
1. ugh.

2. Does this make the update of `pushRules` in `MatrixClient.setRoomMutePushRule` redundant (or racy)?